parsers/FR_O.py

```diff
@@ -0,0 +1,277 @@
+    ],
+    'RE': [
+        {
+            'name': 'bagasse_charbon_mwh',
+            'monthly_variation': True,
+            'type': MAP_GENERATION_BAGASSE_COAL_REUNION
+        },
+        {
+            'name': 'bagasse_charbon',
+            'monthly_variation': True,
+            'type': MAP_GENERATION_BAGASSE_COAL_REUNION
+        },
+        {
+            'name': 'thermique_mwh',
+            'monthly_variation': False,
+            'type': MAP_GENERATION_DIESEL_GAS_REUNION
+        },
+        {
+            'name': 'thermique',
+            'monthly_variation': False,
+            'type': MAP_GENERATION_DIESEL_GAS_REUNION
+        }
+    ],
+}
+
+# Non-thermal sources
+sources = ['biomass', 'hydro', 'solar', 'wind', 'geothermal']
+# Non-thermal sources names in api
+source_mapping = {
+    'biomass': ['bioenergies_mwh', 'bioenergies'],
+    'hydro': ['hydraulique_mwh', 'hydraulique'],
+    'solar': ['photovoltaique_mwh', 'photovoltaique', 'photovoltaique_avec_stockage'],
+    'wind': ['eolien_mwh', 'eolien'],
+    'geothermal': ['geothermie_mwh'],
+    'gas': ['biogaz']
+}
+
+
+def fetch_production(zone_key=None, session=None, target_datetime=None,
+                     logger=logging.getLogger(__name__)):
+    if (target_datetime is None) and (zone_key != 'RE'):
+        raise NotImplementedError('There is no real time data')
+
+    # Request build
+    r = session or requests.session()
+    params = get_param(zone_key, target_datetime)
+    api = get_api(zone_key, target_datetime)
+
+    # Date build
+    if target_datetime is None:
+        datetime = arrow.now(tz=params['timezone'])
+    else:
+        datetime = arrow.get(target_datetime, params['timezone'])
+
+    # Data retrievement
+    response = r.get(api, params=params)
+    data = json.loads(response.content)
+    datapoints = []
+
+    # Response build
+    if(len(data['records']) > 0 and ('fields' in data['records'][0])):
+        fields = data['records'][0]['fields']
+        datetime_result = arrow.get(fields['date_heure']).datetime
+        result = {
+            'zoneKey': zone_key,
+            'datetime': datetime_result,
+            'source': get_source(zone_key, target_datetime),
+            'production': {
+                'nuclear': 0,
+                'biomass': 0,
+                'coal': 0,
+                'gas': 0,
+                'hydro': 0,
+                'oil': 0,
+                'solar': 0,
+                'wind': 0,
+                'geothermal': 0,
+                'unknown': 0
+            }
+        }
+        # Non-thermal sources
+        for i in range(0, len(sources)):
+            current_sources = source_mapping[sources[i]]
+            for j in range(0, len(current_sources)):
+                current_source = current_sources[j]
+                if(current_source in fields):
+                    value = fields[current_source]
+                    if value > 0:
+                        result['production'][sources[i]] += value
```

Comment:
Instead of `for i in range(0, len(sources))` you could do `for source in sources`, it tends to be more readable in that format.

Reply:
Totally. I'm not used to Python, I never worked with this language.
Thank you for the tip.

Reply:
No problem, I found this [Raymond Hettinger talk](https://www.youtube.com/watch?v=OSGv2VnC0go) a great overview of python a while back.